recipes/libmodbus/all/conanfile.py

```diff
@@ -41,16 +39,15 @@ def configure(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("libmodbus-{}".format(self.version), self._source_subfolder)
-
     def build_requirements(self):
         if self.settings.compiler == "Visual Studio":
-            self.build_requires("automake/1.16.1")
-        if tools.os_info.is_windows and "CONAN_BASH_PATH" not in os.environ \
-                and tools.os_info.detect_windows_subsystem() != "msys2":
-            self.build_requires("msys2/20190524")
+            self.build_requires("automake/1.16.3")
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
```

Comment:
Usage here of the `tools.os_info` can lead to errors. If someone computes the package-id using a Linux profile in a Windows machine it will get invalid-config for this build-requires: 
 * as it is running in Windows, it will evaluate `tools.os_info.is_windows` to `True`
 * as profile belongs to `Linux` it will compute the packageID for `msys2/cci.latest` using `os=Linux` and it will fail.
 
Luckily we are doing the opposite in CCI: running in Linux a Windows profile, so `msys2/cci.latest` is not evaluated at that time.

---

I know this is not related to this PR, it is not changing that logic, but today I wrote about the same in another issue and it is worth noticing it here as well.

Reply:
I know, but it requires lot of boilerplate to properly check that in a robust way for 1 profile and 2 profiles. I hope that 1 profile will have self.settings_build soon.

Reply:
there is this little helper

```
def _build_os(self):
    if hasattr(self, "settings_build"):
        return self.settings_build.os
    else:
        return self.settings.os
```

Reply:
or use this helper if you need access to `os/arch`.
```
@python
def _settings_build(self):
    return self.settings_build if hasattr(self, "settings_build") else self.settings
```
Then you can refer to `self._settings_build.os` or `self._settings_build.arch`.

Hopefully, by conan 2.0, we get a `self.settings_build` object unconditionally and will have an easier time refactoring.
This is of course a big if.

Reply:
Ok, let's try in this PR

Reply:
Currently, this will not work everywhere when `os` is not in `self.settings`.

Reply:
How os could not be in settings?

Reply:
For header-only project and such.

Reply:
Uh sure, but we just have to update the recipe. It was recommended to add all settings even for header only lib anyway.

Reply:
Ah, good idea. `self.info.header_only()` clears `self.info.settings` anyways.
Thanks for the hint.